main/src/ca/uwaterloo/flix/language/phase/Lowering.scala

```diff
@@ -540,19 +547,22 @@ object Lowering {
 
     case Expression.NewChannel(exp, tpe, pur, eff, loc) =>
       val e = visitExp(exp)
-      val t = visitType(tpe)
-      Expression.NewChannel(e, t, pur, eff, loc)
+      val unboxedChannelType = Type.mkEnum(Enums.ChannelMpmc, e.tpe :: Nil, loc)
```

Comment:
Oh I see an issue, you use `e.tpe` as the element type, but `e` here is the size expression, always `Int32`. It should be `tpe`